### PR TITLE
Fix powershell commands

### LINKED ISSUE #5032: Commands do not work on UWP projects
This bug was hiding under #4802 . 

### Repro steps

Install Microsoft.EFCore.Commands
Execute any EF command from PMC.

Error:
```
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.AspNetCore.Hosting.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.AspNetCore.Hosting.Abstractions, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
```

I've also seen this one:
```
> Add-Migration First -Verbose
Using startup project 'test\UWP.SQLite'.
Using project 'test\UWP.SQLite'
Build started...
Build succeeded.
Using application base 'C:\Users\namc\dev\ef.platformtesting\test\UWP.SQLite\bin\x86\Debug\'.
No configuration file found.
Using current directory 'C:\Users\namc\dev\ef.platformtesting\test\UWP.SQLite\bin\x86\Debug\'.
System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeAssembly.get_DefinedTypes()
   at Microsoft.EntityFrameworkCore.Design.Internal.StartupInvoker..ctor(Assembly startupAssembly, String environment, String startupProjectDir)
   at Microsoft.EntityFrameworkCore.Design.DbContextOperations..ctor(ILoggerProvider loggerProvider, Assembly assembly, Assembly startupAssembly, String environment, String startupProjectDir)
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations..ctor(AssemblyLoader assemblyLoader, ILoggerProvider loggerProvider, Assembly assembly, Assembly startupAssembly, String environment, String projectDir, String startupProjectDir, String rootNamespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.<>c__DisplayClass4_0.<.ctor>b__5()
   at Microsoft.EntityFrameworkCore.Internal.LazyRef`1.get_Value()

Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
```

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #5032 and #5036 

Depends on #5021 

cc @bricelam @divega

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/StartupInvoker.cs [hunk 1]
```diff
@@ -126,6 +130,18 @@ protected virtual IServiceCollection ConfigureHostServices([NotNull] IServiceCol
                     services.AddSingleton(PlatformServices.Default.Runtime);
                 }
             }
+#endif
+        }
+
+        protected virtual IServiceCollection ConfigureHostServices([NotNull] IServiceCollection services)
+        {
+            if (_aspnetHosting)
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/StartupInvoker.cs]: @natemcmaster and I talked about this and we are going to attempt an approach that should avoid leaking this flag to the public surface. Assuming it works it is still just an RC2 only workaround, but would have less impact.

**Comment by natemcmaster** [comment]
@divega updated. This relies on #5021. To see changes relative only to this PR, see https://github.com/aspnet/EntityFramework/pull/5037/commits/c7c6fa914d0056fd36070885cf547d2520fe20f2.

**Comment by divega** [comment]
I think the comment you had in the previous iteration as well as a TODO comments there and on the DesignTimeEnvironment when we had a more optimal implementation would be good. 

Other than that :shipit:

**Comment by natemcmaster** [comment]
Updated and rebased off latest UWP cross compiling change. 

:watch: will merge this after CI passes #5021

File: src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
 
+// ReSharper disable All
+
+// CAUTION / CAUTION / CAUTION
+//
+// This file needs to be written in C# 5. It is also compiled by the powershell host 
+// in Package Manager Console which does not support C# 6.
+//
+// CAUTION / CAUTION /CAUTION
+
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs]: Lol, silly R#, I wrote it that way for a reason....

**Comment by bricelam** [comment]
:shipit:

**Comment by bricelam** [comment]
Any reason we don't want to just cross compile hosting abstractions to netcore50? At least until the .NET Standard delivers on its promise?

**Comment by natemcmaster** [comment]
I'm avoiding cross compiling hosting since that has a deep dependency chain that spreads. Since we don't need aspnet hosting on UWP, I figured minimize cross-compiling.

**Pull request merged**